### ISSUE TITLE
Road Shader

### ISSUE THREAD

**Issue by Madecu** [carla-simulator/carla#7383]
Enhance the road shaders and textures  to make it more photorealistic:

**Comment by AinaRoca** [comment]
1) The car paint shader is working, it allowed us to make three different types of car paint. We want to add dirt to the car paint but it it is not finished yet, it will be on hold until it becomes a priority.

![Image](https://github.com/carla-simulator/carla/assets/119680845/b7a122ff-ffe3-4d33-bdfe-d8446918681b)
![Image](https://github.com/carla-simulator/carla/assets/119680845/a5bba6c5-697e-410d-b881-11c0758fe12f)
![Image](https://github.com/carla-simulator/carla/assets/119680845/65145d4b-008e-45ee-978f-92e6014243c3)

2) We created on substance painter different generic materials that are needed for the exterior parts of the car. It will make texturing faster.

![Image](https://github.com/carla-simulator/carla/assets/119680845/13188b9c-a17c-4c92-a70b-31d73c2a589d)

That's how it looks like with the new textures:

![Image](https://github.com/carla-simulator/carla/assets/119680845/552cb118-59eb-48cd-a76b-1eef71a3f4af)

3) Now we are working on the base of the asphalt material, once this is done we will start adding other details like cracks.
We will update with an image when the base is more advanced.

**Comment by AinaRoca** [comment]
We decided to change the way we were doing the shader. Instead of making the base and add details to it, we took the matrix asphalt shader, which was what we used to make the base, deleted the things we didn't need and modified the ones we wanted to change, like some parameters and some textures. There's still some work to do regarding cracks and stones, as you can see those rubble are a bit too big.
Once this is done, we will focus on the puddles to fully understand how it works and decide what we keep and what we change or delete. 



![Image](https://github.com/carla-simulator/carla/assets/119680845/3bc6a5b2-f254-45b0-b8a6-7bac5549f541)

**Comment by AinaRoca** [comment]
The shader is almost done, that's how it looks like right now.
Many things can be modified on this shader like cracks, rubble or patches. I will make a complete explantion once it is done.
Here I add come before and after:
![HighresScreenshot00003](https://github.com/carla-simulator/carla/assets/119680845/63b387ed-20d9-4544-b490-3eb98e92f2b6)
![HighresScreenshot00000](https://github.com/carla-simulator/carla/assets/119680845/e246e7bd-7569-4753-bdb9-2d2e7da21b17)
![HighresScreenshot00006](https://github.com/carla-simulator/carla/assets/119680845/bc51469a-5020-4647-a518-b8ecd7824bc4)
![HighresScreenshot00005](https://github.com/carla-simulator/carla/assets/119680845/9f70899f-54e9-4d08-867e-32a738072e95)

![HighresScreenshot00002](https://github.com/carla-simulator/carla/assets/119680845/81d28a54-35db-4a6f-87f0-582075d574ec)
![HighresScreenshot00001](https://github.com/carla-simulator/carla/assets/119680845/405e39a2-b9ac-4aad-b27f-96e6d19a9420)

**Comment by AinaRoca** [comment]
Asphalt shader done, it has changed quite a lot since I said it was almost finished...


![Image](https://github.com/carla-simulator/carla/assets/119680845/2652bace-0e48-430d-bd55-e57645461711)


![Image](https://github.com/carla-simulator/carla/assets/119680845/604b905b-3d0b-4dda-952a-ad0581631384)


![Image](https://github.com/carla-simulator/carla/assets/119680845/3a9ebd8b-4aea-442c-b8f2-b74186e277f8)

The base of the material has two kinds of asphalt that can be controlled with a mask to give it more variation.

Then there are other elements that can be controlled separatedly, with its own textures, roughness, color, size, etc:

-Rubble
-Oil spill
-Grime
-Macro variation
-Patches
-Cracks
-Wheel marks
-Puddles

**Comment by lt-wang** [comment]
How it was achieved
